src/wp-includes/i18n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,404 @@
+<?php
+/**
+ * I18N: WP_Translation_Controller class.
+ *
+ * @package WordPress
+ * @subpackage I18N
+ * @since 6.5.0
+ */
+
+/**
+ * Class WP_Translation_Controller.
+ *
+ * @since 6.5.0
+ */
+class WP_Translation_Controller {
+	/**
+	 * Current locale.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	protected $current_locale = 'en_US';
+
+	/**
+	 * Map of loaded translations per locale and text domain.
+	 *
+	 * [ Locale => [ Textdomain => [ .., .. ] ] ]
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string, array<string, WP_Translation_File[]>>
+	 */
+	protected $loaded_translations = array();
+
+	/**
+	 * List of loaded translation files.
+	 *
+	 * [ Filename => [ Locale => [ Textdomain => WP_Translation_File ] ] ]
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string,array<string, array<string, WP_Translation_File|false>>>
+	 */
+	protected $loaded_files = array();
```

Comment:
Nit-pick: While not mandated by WPCS, per WP coding standards there shouldn't be a blank line between `@since` and `@var` on class properties. See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#2-classes:~:text=2.1%20Class%20Members-,2.1.1%20Properties,-Class%20properties%20should
```suggestion
	/**
	 * Current locale.
	 *
	 * @since 6.5.0
	 * @var string
	 */
	protected $current_locale = 'en_US';

	/**
	 * Map of loaded translations per locale and text domain.
	 *
	 * [ Locale => [ Textdomain => [ .., .. ] ] ]
	 *
	 * @since 6.5.0
	 * @var array<string, array<string, WP_Translation_File[]>>
	 */
	protected $loaded_translations = array();

	/**
	 * List of loaded translation files.
	 *
	 * [ Filename => [ Locale => [ Textdomain => WP_Translation_File ] ] ]
	 *
	 * @since 6.5.0
	 * @var array<string,array<string, array<string, WP_Translation_File|false>>>
	 */
	protected $loaded_files = array();
```